File: controllers/simpleLogger.js
const fs = require("fs");
const path = require("path");
const logModel = require("../models/log");
const zip = new require("node-zip")();
const { Parser } = require('json2csv');

const getAction = (teleporter, object) => object != "" ? object : teleporter

const getAlphabet = (agent, action, object) => agent[0].toUpperCase() + action[0].toUpperCase() + object[0].toUpperCase() + object.slice(-1).toUpperCase()

const jsonToCSV = (logInput) => {   
    const fields = ['type', 'id', 'timestamps.sceneTimer'];   // define the columns here
    const opts = { fields };

    try {
      const parser = new Parser(opts);
      const csv = parser.parse(logInput);
      console.log(csv);
      return csv
    } catch (err) {
      console.error(err);
    }
}

const emptyAction = {
    actionID: "",
    teleporterID: "",
    objectID: "",
    position : {x : 0, y : 0, z : 0}
}

const simpleLogger = (files) => {
    return logModel.find({  name: { $in: files }  }, (err, logs) => {
        const data = logs.map((log) => {
            const entries = log.entries
            if (entries.agentID && entries?.action?.actionID) {   
                const temp = {
                    "Time": "", 
                    "Alphabet": "", 
                    "Action": "", 
                    "State": ""
                }
                    
                if (entries.action) {
                    temp.Actions = `${entries.id} ${entries?.action?.actionID} ${getAction(entries?.action?.objectID)} ${entries?.action?.teleporterID}`
    
                    temp.Alphabet = getAlphabet(entries.id, entries.action.actionID, getAction(entries.action.objectID, entries.action.teleporterID))
                    
                    temp.Time = entries.timestamps.sceneTimer
                }

                if (entries.state) {
                    temp.states = `${entries.agentID} ${entries.id} ${entries.state}`
                }
                return temp
            }
        });
        
        zip.file(log.name + ".json", JSON.stringify(log.entries));
        zip.file(log.name + ".csv", jsonToCSV(data));
        
        const zippedLogs = zip.generate({ base64: false, compression: "DEFLATE" });
        fs.writeFileSync(path.join(__dirname, "logFiles.zip"), zippedLogs, "binary");

        res.download(path.join(__dirname, "logFiles.zip"), (err) => {
        if (err) {
            console.log(err);
        } else {
            //delete the zip file.
            fs.unlinkSync(path.join(__dirname, "logFiles.zip"));
        }
        });
    })
}

module.exports = simpleLogger

File: controllers/simpleLogger.py
# import argparse
import json

# parser = argparse.ArgumentParser()
# parser.add_argument('--logFile', required=True, help='Full Log File')
# args = parser.parse_args()

file = json.load(open(args.logFile))

time = []
actions = []
states = []
alphabet = []
empty_action = {
                    "actionID": "",
                    "teleporterID": "",
                    "objectID": "",
                    "position" : {"x" : 0, "y" : 0, "z" : 0}
                }
            

def get_action(teleporter, object):
    return object if object != "" else teleporter

def get_alphabet(agent, action, object):
    return agent[0].upper() + action[0].upper() + object[0].upper() + object[-1].upper()

for event in file:
    if "agentID" in event and event["agentID"] == "":
        continue

    if "action" in event and event["action"] == empty_action:
        continue

    if "action" in event:
        actions.append("{} {} {}".format( event["id"], event["action"]["actionID"], get_action(event["action"]["objectID"], event["action"]["teleporterID"])))
        alphabet.append(get_alphabet(event["id"], event["action"]["actionID"], get_action(event["action"]["objectID"], event["action"]["teleporterID"])))
        time.append(event["timestamps"]["sceneTimer"])

    if "state" in event:
        states.append("{} {} {}".format(event["agentID"], event["id"], event["state"]))

jsonData = []

for t, al, ac, s in zip(time, alphabet, actions, states):
    d = {"Time" : t, "Alphabet" : al, "Action" : ac, "State" : s}
    jsonData.append(d)


with open('temp.json', 'w') as f:
    json.dump(jsonData, f)